scripts/build/setup/termux_setup_rust.sh

```diff
@@ -28,6 +28,9 @@ termux_setup_rust() {
 	ENV_NAME=${ENV_NAME//-/_}
 	export $ENV_NAME=$CC
 	export TARGET_CFLAGS="$CFLAGS $CPPFLAGS"
+	# This was getting applied for the host build of Rust macros or whatever, so
+	# unset it.
+	unset CFLAGS
```

Comment:
@its-pointless, let me know if this is okay to remove for all crates, as the target flag is set just above and this one was getting picked up by C crates that are built for the host, for macros or whatever, and breaking the build.

Reply:
@xeffyr or @grimler91, any opinion on removing this flag for all crates that use C code? I believe it was wrong to keep this, but I don't use Rust.

Reply:
So, without unsetting you were getting build errors during some host compilation for nushell due to unknown flags?

Looks like a reasonable change, but might be a good idea to test build a couple of rust packages after the flag has been unset

Reply:
Yes, but not unknown, [the CI error showed that the problem was these cross-compilation flags wrongly being applied to the host build (see the compiler command used and resulting error)](https://github.com/termux/termux-packages/runs/1063466031#step:5:1105).

I don't think testing other Rust packages will show anything, as this would have broken them already, just like the previous version of nushell built fine. The problem only came up when nushell started depending on OpenSSL in this version, and compiling that C crate dependency for the host finally surfaced this bug.

I will try building a few Rust packages anyway though, to make sure.

Reply:
I tried compiling the librsvg, ripgrep, and starship packages with this patch applied, no problem. I will look into a fix for the x86_64 build and then merge.